myenumerables.rb

```diff
@@ -0,0 +1,158 @@
+        yield(self[item])
+      end
+    else
+      while index < array.size
+        return to_enum(:my_each) unless block_given?
+
+        yield self[index]
+        index += 1
+        return self
+      end
+    end
+  end
+
+  def my_each_with_index(index = 0)
+    while index < size
+      return to_enum(:my_each_with_index) unless block_given?
+
+      yield self[index], index
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |item| result << item if yield(item) == true }
+    result
+  end
+
+  def my_all?(query = nil)
+    if query.class == Class
+      my_each do |idx|
+        return false unless idx.class == query
+      end
+    elsif query.class == String or query.class == Integer
+      my_each do |idx|
+        return false unless idx == query
+      end
+    elsif query.class == Regexp
+      my_each do |idx|
+        return false unless idx.match(query)
+      end
+    elsif block_given?
+      my_each do |idx|
+        return false unless yield(idx)
+      end
+    elsif query.nil?
+      my_each do |idx|
+        return false if idx === false
+      end
+    else
+      my_each do |idx|
+        return false unless !idx.nil?
+      end
+    end
+    true
+  end
+
+  def my_any?(*query)
+    if block_given?
+      length.times { |item| return true unless yield(item) }
+    elsif query.class == Regexp
+      length.times { |item| return true unless item == query }
+    elsif query.class == Class
+      length.times { |item| return true unless item.class == query }
+    elsif query.class == Numeric or query.class == String
+      length.times { |item| return true unless item == query }
+    elsif query.class == Proc
+      length.times { |item| return true unless yield(item) }
+    else
+      length.times { |item| return true unless item }
+    end
+    false
+  end
+
+  def my_none?(*query)
+    if block_given?
+      length.times { |item| return false unless yield(item) }
+    elsif query.class == Regexp
+      length.times { |item| return false unless item == query }
+    elsif query.class == Class
+      length.times { |item| return false unless item.class == query }
+    elsif query.class == Numeric or query.class == String
+      length.times { |item| return false unless item == query }
+    else
+      length.times { |item| return false unless item }
+    end
+    true
+  end
+
+  def my_count(query = nil)
+    array = self
+    count = 0
+    if block_given?
+      array.length.times { |item| count += 1 if yield(array[item]) }
+    elsif !query.nil?
+      array.length.times { |item| count += 1 if array[item] == query }
+    elsif query.nil?
+      count = array.length
+    end
+    count
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    array = []
+    if block_given?
+      to_a.my_each do |item|
+        array.push(yield(item))
+      end
+    end
+    array
+  end
```

Comment:
```rb
2.6.3 :070 > my_proc = proc {|num| num > 10 } 
 => #<Proc:0x0000000002502ee8@(irb):70> 
2.6.3 :071 > array.my_map(my_proc) {|num| num < 10}
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.3/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.3/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):71
        1: from myenumerables.rb:112:in `my_map'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.3 :072 >
```
That should return a new array with the results of running the proc once for every element in enum.